src/test/java/lambda/part3/exercise/Exercise1.java

```diff
@@ -0,0 +1,84 @@
+package lambda.part3.exercise;
+
+import java.util.function.Function;
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"WeakerAccess", "unused"})
+public class Exercise1 {
+
+    @Test
+    public void mapEmployeesToLengthOfTheirFullNames() {
+        List<Employee> employees = getEmployees();
+        List<Integer> lengths = new ArrayList<>();
+
+        Function<Employee, Person> personExtractor = employee -> employee.getPerson();
+
+        Function<Person, String> fullNameExtractor = person -> person.getFullName();
+
+        Function<String, Integer> stringLengthExtractor = s -> s.length();
+
+        Function<Employee, Integer> fullNameLengthExtractor =
+            employee -> stringLengthExtractor
+                .apply(fullNameExtractor.apply(personExtractor.apply(employee)));
```

Comment:
It's better to use `.andThen(...).andThen(...);`

Reply:
Now andThen(...) in use